.github/workflows/update-charts.yml

```diff
@@ -2,185 +2,269 @@ name: Update helm charts
-          mkdir -p $INSTALL_DIR
-          wget -O $INSTALL_DIR/semver https://github.com/fsaintjacques/semver-tool/raw/3.4.0/src/semver
-          chmod +x $INSTALL_DIR/semver
-          echo $INSTALL_DIR >> $GITHUB_PATH
+          INSTALL_DIR="$HOME"/.semver
+          mkdir -p "$INSTALL_DIR"
+          wget -O "$INSTALL_DIR"/semver https://github.com/fsaintjacques/semver-tool/raw/3.4.0/src/semver
+          chmod +x "$INSTALL_DIR"/semver
+          echo "$INSTALL_DIR" >> "$GITHUB_PATH"
 
       - name: Check if it is a patch update
         id: check_update_type
         run: |
-          OLDVERSION=${{github.event.client_payload.oldVersion}}
-          NEWVERSION=${{github.event.client_payload.version}}
-          REPOSITORY=${{github.event.client_payload.repository}}
+          OLDVERSION="${{ needs.setvariables.outputs.old_version }}"
+          NEWVERSION="${{ needs.setvariables.outputs.version }}"
+          REPOSITORY="${{ needs.setvariables.outputs.repository }}"
 
-          VALID=$(semver validate $OLDVERSION)
+          VALID="$(semver validate $OLDVERSION)"
           if [[ $VALID == "invalid" ]]; then
                   exit 1
           fi
 
-          VALID=$(semver validate $NEWVERSION)
+          VALID="$(semver validate $NEWVERSION)"
           if [[ $VALID == "invalid" ]]; then
                   exit 1
           fi
-          UPDATE_TYPE=$(semver diff $OLDVERSION $NEWVERSION)
+          UPDATE_TYPE="$(semver diff $OLDVERSION $NEWVERSION)"
           PRERELEASE_SUFFIX="-$(semver get prerelease $NEWVERSION)"
           if [[ $PRERELEASE_SUFFIX == "-" ]];then
                 PRERELEASE_SUFFIX=" "
           fi
 
 
-          echo "update_type=$UPDATE_TYPE" >> $GITHUB_OUTPUT
-          echo "repository=$REPOSITORY" >> $GITHUB_OUTPUT
-          echo "prerelease=$PRERELEASE_SUFFIX" >> $GITHUB_OUTPUT
+          echo "update_type=$UPDATE_TYPE" >> "$GITHUB_OUTPUT"
+          echo "repository=$REPOSITORY" >> "$GITHUB_OUTPUT"
+          echo "prerelease=$PRERELEASE_SUFFIX" >> "$GITHUB_OUTPUT"
 
   patch-update:
     name: Patch release updates
     runs-on: ubuntu-latest
     needs:
       - check-update-type
+      - setvariables
     if: needs.check-update-type.outputs.update_type == 'patch' && !endsWith(needs.check-update-type.outputs.repository, 'kwctl')
     permissions:
       contents: write
       pull-requests: write
     steps:
-      - name: Set environment variables
-        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
-        with:
-          script: |
-            core.exportVariable("UPDATECLI_GITHUB_OWNER", context.repo["owner"])
-            core.exportVariable("UPDATECLI_CHART_VERSION", context.payload.client_payload.version)
+      - name: create env vars from needs.setvariables.outputs
+        # This is neeeded because in scripts, we don't have a way to access the `needs` json object.
+        # The env vars defined here will be accesible in the scripts under `process.env.foo`
```

Comment:
It may be possible to drop this step, not save the vars as env vars, and in the script parse the `needs.setvariables.outputs` string into json.
So instead of  `let version = process.env.version` do
```let version = JSON.parse( ' ${{ needs.setvariables.outputs }} ' ).version```.
 But I couldn't get it to work.

Reply:
let's leave it around